apps/site/components/withDownloadArchive.tsx

```diff
@@ -0,0 +1,46 @@
+import { notFound } from 'next/navigation';
+import type { FC } from 'react';
+
+import { getClientContext } from '#site/client-context';
+import provideReleaseData from '#site/next-data/providers/releaseData';
+import {
+  buildReleaseArtifacts,
+  extractVersionFromPath,
+  findReleaseByVersion,
+} from '#site/util/download/archive';
+
+type DownloadArchive = ReturnType<typeof buildReleaseArtifacts>;
+
+type WithDownloadArchiveProps = {
+  children: FC<DownloadArchive>;
+};
+
+/**
+ * Higher-order component that extracts version from pathname,
+ * fetches release data, and provides download artifacts to child component
+ */
+const WithDownloadArchive: FC<WithDownloadArchiveProps> = async ({
+  children: Component,
+}) => {
+  const { pathname } = getClientContext();
+
+  // Extract version from pathname
+  const version = extractVersionFromPath(pathname);
+
+  if (version == null) {
+    return notFound();
+  }
+
+  // Find the release data for the given version
+  const releaseData = provideReleaseData();
+  const release = findReleaseByVersion(releaseData, version);
+
+  const releaseArtifacts = buildReleaseArtifacts(
+    release,
+    version === 'archive' ? release.versionWithPrefix : version
```

Comment:
We shouldn't have this specific version === archive thing. /downloads/archive should be a redirect to /download/archive/latestLTS, and that can be handled by Next.js Middleware or even directly on the page.tsx for /download/archive